stub_uploader/metadata.py

```diff
@@ -289,7 +290,7 @@ def verify_external_req(
 
     if req.name not in EXTERNAL_REQ_ALLOWLIST and not _unsafe_ignore_allowlist:
         raise InvalidRequires(
-            f"Expected dependency {req} to be present in the allowlist"
+            f"Expected dependency {req.name} to be present in the allowlist"
```

Comment:
I think it's often unclear to our contributors which allowlist this is referring to. This maybe isn't much better, but it's a small improvement:
```suggestion
            f"Expected dependency {req.name} to be present in the stub_uploader allowlist"
```